Source/DafnyCore/Options/BoogieOptionBag.cs

```diff
@@ -58,6 +59,32 @@ public static class BoogieOptionBag {
     ArgumentHelpName = "seconds",
   };
 
+  public static readonly Option<int> VerificationErrorLimit =
+    new("--error-limit", () => 5, "Set the maximum number of errors to report (0 for unlimited).");
+
+  public static readonly Option<uint> SolverResourceLimit = new("--resource-limit",
+    @"Specify the maximum resource limit (rlimit) value to pass to Z3. A resource limit is a deterministic
+    alternative to a time limit. The output produced by `--log-format csv` includes the resource use of each proof
+    effort, which you can use to determine an appropriate limit for your program. Multiplied by 1000 before
+    sending to Z3.");
+  public static readonly Option<string> SolverPlugin = new("--solver-plugin",
+    @"Dafny uses Boogie as part of its verification process. This option allows customising that part using a Boogie plugin. More information about Boogie can be found at https://github.com/boogie-org/boogie. Information on how to construct Boogie plugins can be found by looking at the code in https://github.com/boogie-org/boogie/blob/v2.16.3/Source/Provers/SMTLib/ProverUtil.cs#L316");
+
+  public static readonly Option<string> SolverLog =
+    new("--solver-log", @"Specify a file to use to log the SMT-Lib text sent to the solver.") {
+      IsHidden = true
+    };
+
+  public static readonly Option<bool> IsolateAssertions = new("--isolate-assertions", @"Verify each assertion in isolation.");
+
+  public static readonly Option<FileInfo> SolverPath = new("--solver-path",
```

Comment:
I would set `SolverPath`, `SolverOption` and `SolverPlugin` (which I still think should be named BoogiePlugin) to hidden, since using them requires looking up additional documentation outside of Dafny. If there are particular Z3 options that we think users are likely to get a benefit front, I suggest we expose them as first class Dafny options.

Reply:
Yeah, my goal with #3938 is to expose more control over the verification process in a friendly way. But this allows expert users to get that control immediately, if they know they could benefit from it. And it allows us to conveniently experiment with various options to help decide which might be worth exposing in a nicer way.

I agree that `SolverOption` should be hidden, and I've made it that way. I'm less convinced about the two `Path` options, which I think will be used more frequently.

As far as the name of `SolverPlugin`, Boogie plugins can only control how it interfaces with the solver. It doesn't provide any mechanism for them to do anything else. So I still prefer `--solver-plugin`.

Reply:
Thanks